server/src/main/java/org/elasticsearch/rest/ChunkedRestResponseBody.java

```diff
@@ -242,4 +243,53 @@ public void close() {
             }
         };
     }
+
+    static ChunkedRestResponseBody fromMany(ChunkedRestResponseBody first, Iterator<? extends ChunkedRestResponseBody> rest) {
+        return new ChunkedRestResponseBody() {
+            private final String contentType = first.getResponseContentTypeString();
+            private ChunkedRestResponseBody current = first;
+
+            @Override
+            public boolean isDone() {
+                return current == null;
+            }
+
+            @Override
+            public ReleasableBytesReference encodeChunk(int sizeHint, Recycler<BytesRef> recycler) throws IOException {
+                try {
+                    return current.encodeChunk(sizeHint, recycler);
```

Comment:
I haven't double checked this, but I'm worried that this'll send a chunk over the wire no matter how big the reference is. If it's less than the sizeHint we should probably give the next one a chance.